kernel/aster-nix/src/net/socket/util/message_header.rs

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use super::socket_addr::SocketAddr;
+use crate::{prelude::*, util::IoVec};
+
+/// Message header used for sendmsg/recvmsg
+#[derive(Debug)]
+pub struct MessageHeader {
+    pub(in crate::net) addr: Option<SocketAddr>,
+    pub(in crate::net) io_vecs: Box<[IoVec]>,
+    pub(in crate::net) control_message: Option<ControlMessage>,
+}
```

Comment:
I think we should have
```rust
#[derive(Debug)]
pub struct MessageHeader {
    pub(in crate::net) addr: Option<SocketAddr>,
    pub(in crate::net) control_message: Option<ControlMessage>,
}
```
and
```rust
fn sendmsg(&self, io_vecs: &[IoVec], msg_hdr: MessageHeader, flags: SendRecvFlags) -> Result<usize>;
fn recvmsg(&self, io_vecs: &mut [IoVec], flags: SendRecvFlags) -> Result<(usize, MessageHeader)>;

fn sendto(&self, io_vecs: &[IoVec], remote: Option<SocketAddr>, flags: SendRecvFlags) -> Result<usize>;
fn recvfrom(&self, io_vecs: &mut [IoVec], flags: SendRecvFlags) -> Result<(usize, Option<SocketAddr>)>;
```

This way,
 - The input and output arguments of `sendmsg` and `recvmsg` are clearer (`io_vecs` should not be included in the return value of `recvmsg`).
 - The extra heap allocation when implementing `recvfrom` and `sendto` (`vec![user_buf].into_boxed_slice()`) can be avoided.
 - `` FIXME: peer_addr should be `None` for recvmsg `` can be addressed. I don't think the Linux implementation writes the remote address in `recvfrom` (maybe you can write a test case to verify this).